File: oop.js
/* 
Abstracion, Encapsulamiento, Polimorfismo y Herencia
*/

class Person {
    name;
    lastname;
    age;

    constructor(name = "John", lastname = "Doe", age = "Unknow"){
        this.name = name;
        this.lastname = lastname;
        this.age = age;
    }

    comer(){

    }

    correr(){
        this.comer();
        console.log("Person running");
    }
}
class Student extends Person {
    college = null;
    constructor(name, lastname, age, college = "MIT"){
        console.log('Student');
        super(name, lastname, age);
        this.college = college;
    }

    correr(){
        console.log("Student running");
    }
}


let est1 = new Student("Jane", "Doe", 29, "Harvard");

est1.comer()
console.log(est1.name);
console.log(est1.college);
est1.correr();


function saludo(){
    return;
}

File: oop.py
""" 
Abstracion, Encapsulamiento, Polimorfismo y Herencia
"""

class Person:
    name = None
    lastname = None
    age = None

    def __init__(self, name = "John", lastname = "Doe", age = "Unknow"):
        self.name = name
        self.lastname = lastname
        self.age = age

    def comer(self):
        return

    def correr(self):
        print("Person running")


class Student(Person):
    college = None

    def __init__(self, name = "John", lastname = "Doe", age = "Unknow", college = "MIT"):
        super().__init__(name, lastname, age)
        self.college = college

    def correr(self):
        print("Student running")


est1 = Student("Jane", "Doe", 29, "Harvard")
print(est1.name)
print(est1.college);

est1.correr()